### ISSUE TITLE
upload pdf to API instead of download in ngx-extended-pdf-viewer

### ISSUE THREAD

**Issue by JayPatel52970** [stephanrauh/ngx-extended-pdf-viewer#1253]
I like the form editing feature in the latest version. In my case, pdf is coming from a remote server and allows users to edit the forms embedded in the pdf. Instead of downloading, I want that edited pdf to send back to the server.

**Comment by stephanrauh** [comment]
Sounds like a low-hanging fruit. Actually, you can even implement a temporary solution yourself, using the internal API of pdf.js. I've entered this line in the console, and it prints something that looks like a PDF file:

```javascript
PDFViewerApplication.pdfDocument.getData().then((data) => { var blob = new Blob([data], { type: "application/pdf" }); console.log(blob)})
```      

In TypeScript, this becomes this code:

```typescript
function getCurrentPdfFileAsBlob(): Blob {
    const pdfViewerApplication: IPDFViewerApplication = (window as any).PDFViewerApplication;
    const data = await pdfViewerApplication.pdfDocument.getData();
    return new Blob([data], { type: "application/pdf" });
}
```

I haven't tested this code yet (apart from the little smoke test), but I'm 90% sure it downloads the PDF file including the current form values.

**Comment by JayPatel52970** [comment]
hey, @stephanrauh  Thanks for the quick answer.

I tried your code and other solutions as well. All are working in just some pdfs.
Also, I found the below function in NgxExtendedPdfViewerService that works exactly as we wanted.


private printService: NgxExtendedPdfViewerService

const blob = await this.printService.getCurrentDocumentAsBlob();


It is working on just some documents only.
However sometime after editing the form when I press the download or save button, it triggers and downloads the original document instead of the modified PDF.

And I get this warning in the console when it doesn't work as expected. 
![image](https://user-images.githubusercontent.com/31163863/159150020-e7355e50-78ec-410f-a349-1212499c518f.png)

I don't know if this is an issue with the document because the same document is working as expected in the native chrome and firefox viewer.

**Comment by stephanrauh** [comment]
In other words, printing ignores form data, too? That's unexpected. Can you send me one of the broken PDF files?

**Comment by JayPatel52970** [comment]
Yes. @stephanrauh 
I've checked with this PDF.


[MiniMCT and MH+ - NO coversheet.pdf](https://github.com/stephanrauh/ngx-extended-pdf-viewer/files/8330211/MiniMCT.and.MH%2B.-.NO.coversheet.pdf)

**Comment by stephanrauh** [comment]
I trust this is just a test document, without real data?

**Comment by stephanrauh** [comment]
I've edited the "physical interpretation" and both printed and downloaded the document using the UI of https://pdfviewer.net. In both cases, my new text showed. If I get you right, it shows the original text on your machine?

**Comment by stephanrauh** [comment]
I can't reproduce the bug. However, if you're using `[(formData)]`, there might be a race condition. Two-way binding has been frowned upon for a reason, and that might cause your bug. Just a wild shot.

It might help to use the non-minified version of the base library (just add the attribute `[minifiedJSLibraries]="false"`). However, the error message of your screenshot is just a follow-up message, so it'd be interesting to see the root cause.

The warning concerning the font should be a thing of the past now. I've fixed that with #1292.

I'm closing the issue now, but I'm happy to re-open it when you've got new insight or if you send me a reproducer.
